tests/integration/vlan_test.py

```diff
@@ -64,6 +65,22 @@ def test_add_and_remove_vlan(eth1_up):
     assert not current_state[INTERFACES]
 
 
+@pytest.fixture
+def vlan_with_base_iface(eth1_up):
```

Comment:
IMHO the `_with_base_iface` is redundant and can be removed because all vlans have a base iface. Either just `vlan` or `vlan_on_eth1` are my ideas for better names.